modules/tinymce/src/core/test/ts/browser/content/insert/MergeInsertedSpansTest.ts

```diff
@@ -0,0 +1,208 @@
+import { describe, it } from '@ephox/bedrock-client';
+import { Arr } from '@ephox/katamari';
+import { TinyAssertions, TinyHooks, TinySelections } from '@ephox/wrap-mcagar';
+
+import Editor from 'tinymce/core/api/Editor';
+import * as InsertContent from 'tinymce/core/content/InsertContent';
+
+interface SpanStyle {
+  style: string;
+  value: string;
+}
+
+describe('browser.tinymce.core.content.insert.MergeInsertedSpansTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    base_url: '/project/tinymce/js/tinymce',
+  }, [], true);
+
+  const red = '#ff0000';
+  const green = '#00ff00';
+  const blue = '#0000ff';
+
+  // const buildNestedSpans = (style: string, values: string[], text: string): string => Arr.foldr(values, (acc, v) => `<span style="${style}: ${v};">${acc}</span>`, text);
+  const buildNestedSpans = (spans: SpanStyle[][], text: string): string => Arr.foldr(spans, (acc, spanStyles) => `<span style="${Arr.foldr(spanStyles, (acc, span) => ` ${span.style}: ${span.value};${acc}`, '').trim()}">${acc}</span>`, text);
```

Comment:
Could you break this apart having a function call with a predicate lambda defined inside a template string makes it very hard to read.